server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -148,6 +148,7 @@ private static TransportVersion findCurrent() {
                 .map(e -> e.getCurrentTransportVersion(TransportVersions.LATEST_DEFINED))
                 .orElse(TransportVersions.LATEST_DEFINED);
             assert version.onOrAfter(TransportVersions.LATEST_DEFINED);
+            assert version.equals(TransportVersions.BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1) == false : "Not supported";
```

Comment:
We don't need a specific check for that version number in this general-purpose code, 8.x will never see that version.

Reply:
Ok, what's the best way to deal with that? Should I just delete this assert?

Reply:
Yeah, I dont think its needed here